cpp/log/cert.cc

```diff
@@ -1370,12 +1504,31 @@ util::Status CertChain::IsValidSignatureChain() const {
                         "certificate chain is not loaded");
   }
 
+  // Search up the chain from the leaf for a certificate with a DSA
```

Comment:
Seeing as you do all this only to throw the `|dsa_cert_with_params|` away unused, I'd suggest that you just delete this for now and only deal with it when you really have to. You can't even coerce OpenSSL to spit out such a cert without hacking into source so I think it's really unlikely that they exist.